src/App.jsx

```diff
@@ -1,14 +1,58 @@
 import './App.css';
+import messages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.jsx';
+import ColorChoice from './components/ColorChoice.jsx';
 
 const App = () => {
+  const [messageData, setMessage] = useState(messages);
+  const [mainSenderColor, setMainSenderColor] = useState('black');
+  const [remoteSenderColor, setRemoteSenderColor] = useState('black');
+  const MainSender = 'Vladimir';
+  const senderNames = [...new Set(messageData.map(message => message.sender))];
+  const remoteSender = senderNames.filter(sender => sender!== MainSender);
+
+  const likeMessage = (id) => {
+    setMessage(messageData => {
+      return messageData.map(message => {
+        if (message.id == id){
+          return {...message, liked: !message.liked};
+        }else {
+          return message;
+        }
+      });
+    });
+  };
+  const sumLikes = () => {
+    return messageData.filter(message => (message.liked)).length;
+  };
+
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between <span style={{ color: mainSenderColor }}> {MainSender} </span> and <span style={{ color: remoteSenderColor }}> {remoteSender} </span>
+        </h1>
+        <div className='sub-header'>
+          <div className='sender-color'>
+            <h2 style={{ color: mainSenderColor }}>{MainSender}&apos;s color:</h2>
+            <ColorChoice changeColor={setMainSenderColor} />
+          </div>
+          <h2>Total of {sumLikes()} ❤️s</h2>
+          <div className='sender-color'>
+            <h2 style={{ color: remoteSenderColor }}>{remoteSender}&apos;s color:</h2>
+            <ColorChoice changeColor={setRemoteSenderColor} />
+          </div>
+        </div>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog
+          entries = {messageData}
+          likeMessage = {likeMessage}
```

Comment:
Like my comment above, I'd prefer the variable name of this prop to be `onSomething` so that it's immediately evident that the prop's value is an event handler.

```suggestion
         onLikeMessage = {likeMessage}
```